File: beecrowd/1024/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	// não aceito pelo avaliador do beecrowd
	var A int
	fmt.Scan(&A)

	for i := 0; i < A; i++ {
		scanner := bufio.NewScanner(os.Stdin)
		scanner.Scan()
		line := scanner.Bytes()

		for i := 0; i < len(line); i++ {
			if line[i] >= 'A' && line[i] <= 'Z' || line[i] >= 'a' && line[i] <= 'z' {
				line[i] = line[i] + 3
			}
		}

		for i, j := 0, len(line)-1; i < j; i, j = i+1, j-1 {
			line[i], line[j] = line[j], line[i]
		}

		for i, ll := range line[len(line)/2:] {
			line[len(line)/2:][i] = ll - 1
		}

		fmt.Println(string(line))
	}
}

File: beecrowd/1024/main.py
# solução aceita pelo avaliador do beecrowd

A = int(input())

for _ in range(A):
  line = list(input())

  for i in range(len(line)):
    if 'A' <= line[i] <= 'Z' or 'a' <= line[i] <= 'z':
      line[i] = chr(ord(line[i]) + 3)

  i, j = 0, len(line) - 1
  while i < j:
    line[i], line[j] = line[j], line[i]
    i, j = i + 1, j - 1

  for i in range(len(line) // 2, len(line)):
    line[i] = chr(ord(line[i]) - 1)

  print(''.join(line))
